examples/chat_completion/chat_completion_functions_rest.py

```diff
@@ -0,0 +1,160 @@
+from os import environ
+import json
+import requests
+
+from mona_openai import get_rest_monitor
+
+MONA_API_KEY = environ.get("MONA_API_KEY")
+MONA_SECRET = environ.get("MONA_SECRET")
+MONA_CREDS = {
+    "key": MONA_API_KEY,
+    "secret": MONA_SECRET,
+}
+
+# This is the name of the monitoring class on Mona
+MONITORING_CONTEXT_NAME = "MONITORED_CHAT_COMPLETION_USE_CASE_NAME"
+# Get Mona logger
+mona_logger = get_rest_monitor(
+    "ChatCompletion",
+    MONA_CREDS,
+    MONITORING_CONTEXT_NAME,
+)
+
+# Set up the API endpoint URL and authentication headers
+url = "https://api.openai.com/v1/chat/completions"
+headers = {
+    "Content-Type": "application/json",
+    "Authorization": f"Bearer {environ.get('OPEN_AI_KEY')}",
+}
+
+
+def get_current_weather(location, unit="fahrenheit"):
+    """Get the current weather in a given location"""
+    weather_info = {
+        "location": location,
+        "temperature": "72",
+        "unit": unit,
+        "forecast": ["sunny", "windy"],
+    }
+    return json.dumps(weather_info)
+
+
+def run_conversation():
+    messages = [
+        {"role": "user", "content": "What's the weather like in Boston?"}
+    ]
+    functions = [
+        {
+            "name": "get_current_weather",
+            "description": "Get the current weather in a given location",
+            "parameters": {
+                "type": "object",
+                "properties": {
+                    "location": {
+                        "type": "string",
+                        "description": "The city and state, e.g. San "
+                        "Francisco, CA",
+                    },
+                    "unit": {
+                        "type": "string",
+                        "enum": ["celsius", "fahrenheit"],
+                    },
+                },
+                "required": ["location"],
+            },
+        }
+    ]
+
+    # Set up the request data
+    data = {
+        "messages": messages,
+        "functions": functions,
+        "max_tokens": 20,
+        "temperature": 0.2,
+        "model": "gpt-3.5-turbo-0613",
+        "n": 1,
+    }
+
+    # The log_request function returns two other function for later logging
+    # the response or the exception. When we later do that, the logger will
+    # actually calculate all the relevant metrics and will send them to
+    # Mona.
+    response_logger, exception_logger = mona_logger.log_request(
+        data, additional_data={"customer_id": "A531251"}
+    )
+
+    try:
+        # Send the request to the API
+        response = requests.post(url, headers=headers, json=data)
+
+        # Check for HTTP errors
+        response.raise_for_status()
+
+        # Log response to Mona
+        response_logger(response.json())
+        print(response.json()["choices"][0]["message"])
```

Comment:
Maybe init a "response_message" var outside the try-catch block with None value and assign it here, your'e repeating this  in line 101